src/lib/y2packager/product_finder.rb

```diff
@@ -142,6 +90,88 @@ def display_order(solvable)
       order
     end
 
+    #
+    # Evaluate the products
+    #
+    # @param product_solvable [Solv] the product solvable to create
+    # @param base_products [Array<String>] the found base products
+    #
+    # @return [Array<Hash>] the found products
+    #
+    def create_products(product_solvable, found_base_products, selected_base)
+      # the dependent repositories, includes also the transient dependencies
+      required = []
+      solver = pool.Solver
+      jobs = select_products(product_solvable, selected_base)
+
+      # run the solver to evaluate all dependencies
+      problems = solver.solve(jobs)
+
+      # if the solver failed we cannot evaluate the dependencies,
+      # something is probably missing or there are conflicts
+      if problems.empty?
+        # find all repositories which have a product selected to install
+        solver.transaction.newsolvables.each do |new_solvable|
+          next if new_solvable == product_solvable
+          new_solvable.lookup_deparray(Solv::SOLVABLE_PROVIDES).each do |dep|
+            next unless dep.str.start_with?("product(")
+            required << new_solvable.repo.name
+          end
+        end
+
+        required.uniq!
+        required.sort!
+      end
```

Comment:
:+1:  for moving to own method, but what about moving also this computation of requirements to own method that can also return nil. So in the end it can look like `depends_on:      required(product_solvable, selected_base)` and whole first part of method can live there.